gnrc_tcp: improved option handling

brummer-simon opened the pull request:
- Removal of the criticized Option Field in the gnrc_tcp_hdr
- Header Cleanup

OlegHahm commented:
Seems reasonable to me.

smlng commented on sys/include/net/gnrc/tcp/hdr.h:
```diff
@@ -29,24 +29,18 @@ extern "C" {
+    network_uint16_t window;      /**< window, in network byte order */
+    network_uint16_t checksum;    /**< checksum, in network byte order */
+    network_uint16_t urgent_ptr;  /**< urgent pointer, in network byte order */
+} gnrc_tcp_hdr_t;
```
I'd rather leave it `tcp_hdr_t` and further move the header (similar to udp) to `sys/include/net/tcp.h`, as I already did in #6509. However, I agree to removing the options array.

smlng replied:
because the header is **not** GNRC specific, hence it should not reside exclusively inside GNRC.

smlng commented on sys/net/gnrc/transport_layer/tcp/gnrc_tcp_option.c:
```diff
@@ -33,58 +33,52 @@ uint16_t _option_build_offset_control(uint16_t nopts, uint16_t ctl)
-        }
+    /* Parse Options Byte by Byte */
+    while (opt_left > 0) {
+        uint8_t kind = *opt_ptr;
```
I would recommend to introduce option struct as follows:

```
struct tcp_opt {
    uint8_t kind,
    uint8_t len,
    uint8_t *value,
} tcp_opt_t;
```

smlng commented:
please consider #6511

brummer-simon commented:
Hi smlng, 

I have moved the tcp header to net/tcp.h and added your option parsing struct to net/tcp.h. The _parse_option Function uses the option struct now. 

Additionally I a bit of cleanup

miri64 commented:
The last commit seems unrelated? Why no new PR for that?

brummer-simon commented:
Jeah I looked at the Files Section in this PR an will split this pr into two seperate pull requests tomorrow during Hack'n'Ack (I might need some help from the more experienced git users)
